### PR TITLE
Feature 1371 custom messages for products

### LINKED ISSUE #1370: Provide user messages inside PDS launcher scripts
## Situation
We have #806 we have the possibility to show up sechub messages. But at PDS side, we are not able to set a user message
And with #976 we provide this in job status  as well.

## Wanted
The PDS launcher script shall have possiblity to send user messages back - e.g. to explain problems etc.

### General
- we need #1371 to be implemented before, so storage and merge in sereco is available
- PDSExecutionCallable: read the message files like done for output and error files
- PDSWorkspaceService: Introduce `messages` folder 
- fetch messages after PDS execution at sechub side by REST API
- pds product executor must store the messages inside the result as json
- Introduce and inject ENV variable `PDS_JOB_USER_MESSAGES_FOLDER` for launcher scripts (inside `PDSLauncherScriptEnvironmentConstants`)

### Inside launcher scripts

We introduce a new environment variable `PDS_JOB_USER_MESSAGES_FOLDER`.
The variable will point to 
`$workspaceLocation/output/messages`

The script can write the user messages as files into this folder. Each user message will have its own file.
- When the file starts with `ERROR_` the corresponding `SecHubMessage` will be of type `error`.
- When the file starts with `WARNING_` the corresponding `SecHubMessage` will be of type `warning`.
- Otherwise the the corresponding `SecHubMessage` will be of type `info`.

The created files may contain multi line text as well. So we can use also tool output as messages!

We will provide a bash snippet example in our documentation to create unique file names:

```bash
#!/bin/bash
# You can paste this snippet inside your launcher script and use it to handle messages

function writeUniqueMessageFile(){
      MSG_PREFIX=$1
      MESSAGE=$2

      MESSAGE_FILE_PATH="${PDS_JOB_USER_MESSAGES_FOLDER}/${MSG_PREFIX}message_$(date +%Y-%m-%d_%H.%M.%S_%N).txt"
      echo "$MESSAGE" > "$MESSAGE_FILE_PATH"
}

function infoMessage(){
    writeUniqueMessageFile "INFO_" "$1"
}

function warnMessage(){
    writeUniqueMessageFile "WARNING_" "$1"
}

function errorMessage(){
    writeUniqueMessageFile "ERROR_" "$1"
}

```
The messages will look like: `message_2022-06-07_18.24.27_691720921.txt`

Usage:
```bash
 infoMessage "this is an info message"
 warnMessage "this is a warning message"
 errorMessage "this is an error message
     with multiple lines... 
 "
```
The former call did create following message files which contain the given text parts: 

 ```bash
    $job_folder_workspace/output/messages
    ├── ERROR_message_2022-06-24_17.56.52_822554054.txt
    ├── INFO_message_2022-06-24_17.56.52_818872869.txt
    └── WARNING_message_2022-06-24_17.56.52_820825342.txt
 ```

### LINKED ISSUE #1371: Products shall be able to add custom messages
## Situation
Currently we do only provide sechub messages (done with #806 ) which are generic - means when a job has a failure or is canceled by user we add the sechub message at the end to the model.

But we cannot provide dedicated messages from products (as an example: owasp zap wrapper could find out that a client certificate is no longer valid and could give the information back as a user message...).

## Wanted
A product executor shall be able to add product specific sechub messages - those shall appear in final report. So it is a kind of 
back channel from products to user.

### Persistence
The product result entity must have a new column for messages (like done for scheduler job status in #976)
`Sereco` shall fetch these information while getting the other product results and add it as `Sereco` annotation so it is available in final report as well (combined)

### LINKED ISSUE #1438: Enhance adapter meta data for PDS adpater with PDS_JOB_UUID
## Situation
We have currently no information inside scan logs which PDS jobs have been run

## Wanted
Stored adapter metadata  shall contain pds job UUID

## Solution
Store the PDS job uuid inside PDS adapter via adapter callback into metadata

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #1371 
- closes #1370 
- closes #1438

### CODE REVIEW

**Review by winzj — APPROVED** [review]
Looks good to me. 
EDIT: I noticed something too late after I already clicked on "approve". Please see the comment below for more information.

**Review by winzj — CHANGES_REQUESTED** [review]
I looked at the files again and noticed one thing the might need to be changed. Please have a look at my comment.

File: sechub-server/src/main/resources/db/migration/U18__add_messages_to_product_result2.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-server/src/main/resources/db/migration/U18__add_messages_to_product_result2.sql]: I found the problems also at U12 and U14. Renamed those files as well in 18e53c4b865b21663e7479fddd5ab044a5b692e9

**Review by winzj — APPROVED** [review]
Looks good to me. Thanks for renaming the other files as well.

**Pull request merged**